Minor feature update

trojanobelix opened the pull request:
Add/extend some exception catches to provide more information what was wrong
Show filename on MouseHover in TabControl

trojanobelix committed:
Merge branch 'pr/n22_fix-workflow' into feature

trojanobelix committed:
Merge branch 'pr/n19_bugfix-array-of-strings-export' into feature

uniederer committed:
Extend dirty-notification with an option to save instantly rather than just falling back

uniederer committed:
Make message more explicit

uniederer committed:
Add autosave option

trojanobelix committed:
Merge branch 'pr/n44_feature/allow_instant_save' into feature

trojanobelix committed:
Merge branch 'feature' of https://github.com/CANopenNode/CANopenEditor into feature

trojanobelix committed:
Merge branch 'bugfix' into feature

trojanobelix committed:
Merge branch 'bugfix' into feature

trojanobelix committed:
Merge branch 'bugfix' into feature

trojanobelix committed:
Bugfix file filter when opening Network XML

nimrof committed:
added it back

trojanobelix committed:
Add/extend some exception catches to provide more information what was wrong

Show filename on MouseHover in TabControl

trojanobelix committed:
Merge remote-tracking branch 'Github_origin/main' into feature

# Conflicts:
#	EDSEditorGUI/DeviceODView.Designer.cs
#	EDSEditorGUI/DeviceODView.cs
#	libEDSsharp/CanOpenNodeExporter.cs

trojanobelix committed:
Merge remote-tracking branch 'Github_origin/ExeptionCatch' into feature

trojanobelix committed:
Unified header for Exporter legacy/V4

trojanobelix committed:
Merge remote-tracking branch 'trojanobelix/feature' into ExeptionCatch

nimrof reviewed (changes requested):
Looks good, just a few comments and i to me the change here introduces a bug.
If you create a device then remove it save in the menu is not disabled
https://github.com/CANopenNode/CANopenEditor/blob/0c97545d570c2560a6bb13beaf8c8ccc32cc0d10/EDSEditorGUI/Form1.Designer.cs#L299


My nitpicking brain may also want to revert the changes in tabControl1_MouseClick as there is no actual change AFIK, but feel free to ignore that

trojanobelix committed:
Merge remote-tracking branch 'Github_origin/main' into feature

trojanobelix committed:
Merge branch 'trojanobelix_main' into feature

trojanobelix committed:
Merge branch 'added-moduleInfo-tab-back-to-DeviceView' into feature

trojanobelix committed:
Update EDSEditorGUI/Form1.cs

Co-authored-by: Lars Elgtvedt Susaas <lars.elgtvedt.susaas@brunvoll.no>

trojanobelix committed:
Update EDSEditorGUI/Form1.cs

Co-authored-by: Lars Elgtvedt Susaas <lars.elgtvedt.susaas@brunvoll.no>

trojanobelix committed:
delete old code fragments

trojanobelix committed:
Merge remote-tracking branch 'trojanobelix/feature' into feature

trojanobelix committed:
Merge remote-tracking branch 'origin/main' into feature

ckhardin committed:
edssharp: add another extension for the EDS file for CountLabel

The count label is required to compile against the CANopenNode v4
library since the definitions are checked by code like the following
in CANopen.c

    #if OD_CNT_NMT != 1
     #error OD_CNT_NMT from OD.h not correct!
    #endif

This problem is exhibited when there is no xpd file that has the
custom properties and there is only an EDS file available. The
StorageLocation is an example of setting the custom property for
the CO_storageGroup and this is adding a similar extension for
the CO_countLabel.

An example eds for NMT 0x1000 is shown below
    [1000]
    ParameterName=Device type
    ObjectType=0x7
    ;CountLabel=NMT
    ;StorageLocation=ROM
    DataType=0x0007
    AccessType=ro
    DefaultValue=0x00000000
    PDOMapping=0

Signed-off-by: Charles Hardin <charles.hardin@redoxblox.com>

trojanobelix committed:
Bugfix for #181

trojanobelix committed:
Merge branch 'pr/n181_clonig_crash' into feature

trojanobelix committed:
Merge remote-tracking branch 'CanopenEditor/feature' into feature

trojanobelix committed:
Merge remote-tracking branch 'Github_origin/main' into feature

trojanobelix committed:
Merge branch 'pr/n180_ckhardin/add-countlabel-eds-extension' into feature

trojanobelix committed:
Merge remote-tracking branch 'Github_origin/main' into feature

trojanobelix committed:
Increment number in object name when adding new objects

trojanobelix committed:
Merge branch 'pr/198' into feature

nimrof commented:
@trojanobelix That last commit changed a lot, do you have a list of the changes?

trojanobelix committed:
Add IncrementNumberAtEnd

Add sub index  in OD: Auto increment sub index name if the name has a number at the end

trojanobelix committed:
Merge branch 'IncObjectName' into feature

# Conflicts:
#	libEDSsharp/eds.cs

trojanobelix committed:
Updates nuget extensions

trojanobelix committed:
Update sourcegrid for .net 8/9

trojanobelix committed:
Check if the sub index for PDO mapping is still valid (might be deleted)

Prevent crash if a mapped sub was deleted from OD.

trojanobelix committed:
Merge branch 'updt_sourcegrid' into feature

trojanobelix committed:
Merge branch 'feat-chkPDOMappingSubValid' into feature

trojanobelix committed:
Prevent the mapping of objects with a data length of 0 bytes

github-actions[bot] committed:
[ROOT][DOCS] contrib-readme-action has updated readme

trojanobelix committed:
Merge remote-tracking branch 'Github_origin/contributors-readme-action-fvt5WTjtvR' into main_RC

trojanobelix committed:
Refactoring

Suppress Warning CS8981
Add comments

trojanobelix committed:
Refractoring

Update Packages

trojanobelix committed:
Refactoring

trojanobelix committed:
refactoring

update gitignore

trojanobelix committed:
Add comment for bug #221

With the switch to .NET 8, System.Drawing is deprecated and no longer included. So it has to be added via NuGet.
Add <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies> to get all dll from NuGet in the build output

trojanobelix committed:
Merge branch 'main_RC' into feature

trojanobelix committed:
bugfix merge artefacts

trojanobelix committed:
bugfix merge artefacts

dependabot[bot] committed:
Bump actions/checkout from 6 to 7

Bumps [actions/checkout](https://github.com/actions/checkout) from 6 to 7.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v6...v7)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

nimrof committed:
automapper upgrade

github-actions[bot] committed:
[ROOT][DOCS] contrib-readme-action has updated readme

nimrof committed:
removed the need for non-digit before the digit in the regex

github-actions[bot] committed:
[ROOT][DOCS] contrib-readme-action has updated readme

trojanobelix committed:
Merge remote-tracking branch 'origin/main' into main_RC

trojanobelix committed:
Merge remote-tracking branch 'origin/automapper-upgrade-v10-and-v13-to-v16' into main_RC

trojanobelix committed:
Merge remote-tracking branch 'origin/228-cloning-an-object-ending-with-a-number-create-an-exception' into main_RC

trojanobelix committed:
Merge remote-tracking branch 'origin/dependabot/github_actions/actions/checkout-7' into main_RC

trojanobelix committed:
Update NuGet

trojanobelix committed:
Merge branch 'main_RC' into feature

trojanobelix committed:
Merge branch 'feature' of https://github.com/trojanobelix/CANopenEditor into feature

# Conflicts:
#	EDSEditorGUI/Form1.cs
#	EDSEditorGUI2/EDSEditorGUI2.csproj
#	GUITests/GUITests.csproj
#	Tests/Tests.csproj
#	libEDSsharp/DocumentationGenMarkup.cs
#	libEDSsharp/libEDSsharp.csproj

github-actions[bot] committed:
[ROOT][DOCS] contrib-readme-action has updated readme
